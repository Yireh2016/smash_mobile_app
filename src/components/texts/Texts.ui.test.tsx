import * as React from 'react';
import renderer from 'react-test-renderer';
import {
  MainTitle,
  H1,
  H2,
  H3,
  SmashLink,
  MainParagraph,
  P1,
  P2,
  P3,
} from './Texts';

describe('Texts renders correctly', () => {
  it('Text H1 renders correctly', () => {
    const Element = () => {
      return <H1>Test</H1>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('Text H2 renders correctly', () => {
    const Element = () => {
      return <H2>Test</H2>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('Text H3 renders correctly', () => {
    const Element = () => {
      return <H3>Test</H3>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('Text P1 renders correctly', () => {
    const Element = () => {
      return <P1>Test</P1>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Text MainTitle renders correctly', () => {
    const Element = () => {
      return <MainTitle>Test</MainTitle>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Text SmashLink renders correctly', () => {
    const Element = () => {
      return <SmashLink>Test</SmashLink>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Text P2 renders correctly', () => {
    const Element = () => {
      return <P2>Test</P2>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  it('Text P3 renders correctly', () => {
    const Element = () => {
      return <P3>Test</P3>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Text MainParagraph renders correctly', () => {
    const Element = () => {
      return <MainParagraph>Test</MainParagraph>;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
