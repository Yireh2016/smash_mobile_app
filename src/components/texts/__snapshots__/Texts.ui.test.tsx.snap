// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texts renders correctly Text H1 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Bold",
      "fontSize": 24,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text H2 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Bold",
      "fontSize": 20,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text H3 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Bold",
      "fontSize": 18,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text MainParagraph renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Regular",
      "fontSize": 18,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text MainTitle renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Bold",
      "fontSize": 28,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text P1 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Regular",
      "fontSize": 16,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text P2 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Regular",
      "fontSize": 14,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text P3 renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#fff",
      "fontFamily": "OpenSans-Regular",
      "fontSize": 12,
    }
  }
>
  Test
</Text>
`;

exports[`Texts renders correctly Text SmashLink renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#29b782",
      "fontFamily": "OpenSans-Bold",
      "fontSize": 16,
    }
  }
>
  Test
</Text>
`;
